services/provider/server/server.go

```diff
@@ -206,3 +217,101 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var j []externalSC.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap(monConfigMap, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s configMap. %v", monConfigMap, err)
+	}
+
+	// This is done to get address of first mon from the monConfigMap configmap
+	cmData := strings.Split(configmap.Data["data"], ",")
+	// Address of first mon
+	monAddr := cmData[0]
+	data := map[string]string{
+		"data":     monAddr,
+		"maxMonId": "0",
+		"mapping":  "{}",
+	}
+	name := monConfigMap
+	kind := "ConfigMap"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing cluster mon.admin key, fsid and name
+	scMon, err := s.consumerAPI.GetSecret(monSecret, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s secret. %v", monSecret, err)
+	}
+	data = map[string]string{
+		"fsid":         string(scMon.Data["fsid"]),
+		"mon-secret":   "mon-secret",
+		"admin-secret": "admin-secret",
+	}
+	name = monSecret
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Ceph user secret
+	storageConsumer, err := s.consumerAPI.GetStorageConsumer(consumerResource.Name, s.namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get storageConsumer resource for the given consumer: %v", err)
+	}
+	cephUser := storageConsumer.Status.CephObjects.CephUser
+	scCephUser, err := s.consumerAPI.GetSecret(fmt.Sprintf("rook-ceph-client-%s", cephUser), s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-client-%s` secret. %v", cephUser, err)
+	}
+
+	data = map[string]string{
+		"userID":  "client." + cephUser,
```

Comment:
instead here , cephUser details need to be added in the provisioner and node secrets